"ScnTool" task failed unexpectedly with Collada asset

<!-- 🚨 STOP 🚨 𝗦𝗧𝗢𝗣 🚨 𝑺𝑻𝑶𝑷 🚨 -->


### Steps to Reproduce

From https://developercommunity.visualstudio.com/content/problem/242099/scntool-task-fails-with-nre-setting-sdkdevpath-cor.html


1. Open attached project.
2. Build.

### Expected Behavior

Not build error.

### Actual Behavior

```
Build FAILED.

/Library/Frameworks/Mono.framework/External/xbuild/Xamarin/iOS/Xamarin.iOS.Common.targets(1163,3): error MSB4018: The "ScnTool" task failed unexpectedly.
/Library/Frameworks/Mono.framework/External/xbuild/Xamarin/iOS/Xamarin.iOS.Common.targets(1163,3): error MSB4018: System.NullReferenceException: Object reference not set to an instance of an object
/Library/Frameworks/Mono.framework/External/xbuild/Xamarin/iOS/Xamarin.iOS.Common.targets(1163,3): error MSB4018:   at Xamarin.MacDev.Tasks.ScnToolTaskBase.SetEnvironmentVariable (System.String variableName, System.String value) [0x00031] in <e8ef46c8f37f4e5da60d8c2600f82452>:0 
/Library/Frameworks/Mono.framework/External/xbuild/Xamarin/iOS/Xamarin.iOS.Common.targets(1163,3): error MSB4018:   at Xamarin.MacDev.Tasks.ScnToolTaskBase.set_SdkDevPath (System.String value) [0x00008] in <e8ef46c8f37f4e5da60d8c2600f82452>:0 
/Library/Frameworks/Mono.framework/External/xbuild/Xamarin/iOS/Xamarin.iOS.Common.targets(1163,3): error MSB4018:   at (wrapper managed-to-native) System.Reflection.MonoMethod.InternalInvoke(System.Reflection.MonoMethod,object,object[],System.Exception&)
/Library/Frameworks/Mono.framework/External/xbuild/Xamarin/iOS/Xamarin.iOS.Common.targets(1163,3): error MSB4018:   at System.Reflection.MonoMethod.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00032] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/corlib/System.Reflection/MonoMethod.cs:305 
/Library/Frameworks/Mono.framework/External/xbuild/Xamarin/iOS/Xamarin.iOS.Common.targets(1169,4): error MSB4026: The "SdkDevPath=$(_SdkDevPath)" parameter for the "ScnTool" task is invalid.
/Library/Frameworks/Mono.framework/External/xbuild/Xamarin/iOS/Xamarin.iOS.Common.targets(1163,3): error MSB4063: The "ScnTool" task could not be initialized with its input parameters. 
    0 Warning(s)
    3 Error(s)

Time Elapsed 00:00:02.13
```

### Environment

<!--
1.
Visual Studio: Help > About Microsoft Visual Studio > Copy Info [button]
Visual Studio for Mac: Visual Studio > About Visual Studio > Show Details > Copy Information [button]

2. Paste into the code block below (between ```)
-->

```
=== Visual Studio Enterprise 2017 for Mac (Preview) ===

Version 7.5 Preview (7.5 build 1161)
Installation UUID: 00d602ef-271b-479b-9054-0433b5797215
Runtime:
	Mono 5.12.0.226 (2018-02/9824e260f56) (64-bit)
	GTK+ 2.24.23 (Raleigh theme)
	Xamarin.Mac 4.4.0.36 (master / 0c7c49a6)

	Package version: 512000226

=== NuGet ===

Version: 4.3.1.4445

=== .NET Core ===

Runtime: /usr/local/share/dotnet/dotnet
Runtime Versions:
	2.0.5
	2.0.0-preview2-25407-01
	1.1.2
	1.0.5
SDK: /usr/local/share/dotnet/sdk/2.1.4/Sdks
SDK Versions:
	2.1.4
	2.0.0-preview2-006497
	1.0.4
MSBuild SDKs: /Library/Frameworks/Mono.framework/Versions/5.12.0/lib/mono/msbuild/15.0/bin/Sdks

=== Xamarin.Profiler ===

Version: 1.6.2
Location: /Applications/Xamarin Profiler.app/Contents/MacOS/Xamarin Profiler

=== Xamarin.Android ===

Version: 8.3.0.17 (Visual Studio Enterprise)
Android SDK: /Users/vidondai/Library/Developer/Xamarin/android-sdk-macosx
	Supported Android versions:
		2.3 (API level 10)
		4.3 (API level 18)
		6.0 (API level 23)
		7.1 (API level 25)
		8.0 (API level 26)
		8.1 (API level 27)

SDK Tools Version: 25.2.5
SDK Platform Tools Version: 25.0.4
SDK Build Tools Version: 23.0.2

Java SDK: /usr
java version "1.8.0_101"
Java(TM) SE Runtime Environment (build 1.8.0_101-b13)
Java HotSpot(TM) 64-Bit Server VM (build 25.101-b13, mixed mode)

Android Designer EPL code available here:
https://github.com/xamarin/AndroidDesigner.EPL

=== Xamarin Inspector ===

Version: 1.4.0
Hash: b3f92f9
Branch: master
Build date: Fri, 19 Jan 2018 22:00:34 GMT
Client compatibility: 1

=== Apple Developer Tools ===

Xcode 9.3 (14154)
Build 9E145

=== Xamarin.Mac ===

Version: 4.5.0.285 (Visual Studio Enterprise)

=== Xamarin.iOS ===

Version: 11.11.0.285 (Visual Studio Enterprise)
Hash: 34c1e4309
Branch: issue-4030
Build date: 2018-05-04 14:40:11-0400

=== Build Information ===

Release ID: 705001161
Git revision: 086c5cc9cf20e6559540a05385a8604b1cf5bf48
Build date: 2018-04-12 18:02:49-04
Xamarin addins: ca9bc3575ea53c0940a34569fd1630a0df96bb2f
Build lane: monodevelop-lion-d15-7

=== Operating System ===

Mac OS X 10.13.4
Darwin 17.5.0 Darwin Kernel Version 17.5.0
    Mon Mar  5 22:24:32 PST 2018
    root:xnu-4570.51.1~1/RELEASE_X86_64 x86_64

=== Enabled user installed addins ===

AddinMaker 1.4.2
MSBuild Editor 2.0.3
NuGet Package Explorer 0.2
Internet of Things (IoT) development (Preview) 7.1


```

### Build Logs

<!--
1. Place cursor below this comment block.

2. Attach build log or link to gist (https://gist.github.com/) of the log.
-->
https://gist.github.com/VincentDondain/4221cb0587494471da07f084fd2ad0e3

### Example Project (If Possible)

<!--
1. Place cursor below this comment block.

2. Drag and drop the compressed project or files needed to reproduce.
-->

[VerticalPlanes_CSharp.zip](https://github.com/xamarin/xamarin-macios/files/1975788/VerticalPlanes_CSharp.zip)

<!--
Switch to the "Preview" tab to ensure your issue renders correctly.
-->

--- a/msbuild/Xamarin.MacDev.Tasks.Core/Tasks/ScnToolTaskBase.cs
+++ b/msbuild/Xamarin.MacDev.Tasks.Core/Tasks/ScnToolTaskBase.cs
@@ -58,16 +58,21 @@ namespace Xamarin.MacDev.Tasks
 			var envVariables = EnvironmentVariables;
 			var index = -1;
 
-			for (int i = 0; i < envVariables.Length; i++) {
-				if (envVariables [i].StartsWith (variableName + "=", StringComparison.Ordinal)) {
-					index = i;
-					break;
+			if (envVariables == null) {
+				envVariables = new string [1];
+				index = 0;
+			} else {
+				for (int i = 0; i < envVariables.Length; i++) {
+					if (envVariables [i].StartsWith (variableName + "=", StringComparison.Ordinal)) {
+						index = i;
+						break;
+					}
 				}
-			}
 
-			if (index < 0) {
-				Array.Resize<string> (ref envVariables, envVariables.Length + 1);
-				index = envVariables.Length - 1;
+				if (index < 0) {
+					Array.Resize<string> (ref envVariables, envVariables.Length + 1);
+					index = envVariables.Length - 1;
+				}
 			}
 
 			envVariables [index] = string.Format ("{0}={1}", variableName, value);

